Moq.AutoMock/AutoMocker.cs

```diff
@@ -360,18 +349,47 @@ public ISetup<TService> Setup<TService>(Expression<Action<TService>> setup)
         /// </summary>
         public void Combine(Type type, params Type[] forwardTo)
         {
-            var mockObject = new MockInstance(CreateMockOf(type));
-            forwardTo.Aggregate(mockObject.Mock, As);
+            if (!(Resolve(type) is MockInstance mockObject))
+                throw new ArgumentException($"{type} did not resolve to a Mock", nameof(type));
 
+            forwardTo.Aggregate(mockObject.Mock, As);
             foreach (var serviceType in forwardTo.Concat(new[] { type }))
                 typeMap[serviceType] = mockObject;
+
+            Mock As(Mock mock, Type forInterface)
+            {
+                var method = mock.GetType().GetMethods().First(x => x.Name == nameof(Mock.As))
+                    .MakeGenericMethod(forInterface);
+                return (Mock)method.Invoke(mock, null);
+            }
+        }
+
+        #endregion Combine
+
+        #region Verify
+
+        /// <summary>
+        /// This is a shortcut for calling `mock.VerifyAll()` on every mock that we have.
+        /// </summary>
+        public void VerifyAll()
+        {
+            foreach (var pair in typeMap)
+            {
+                if (pair.Value.IsMock)

```

Comment:
I realize this matches previous code. But why do we need this `IsMock` if we are going to do a blind type cast. Just use pattern matching and remove the bool property.